react-app/src/features/package/index.tsx

```diff
@@ -54,16 +51,32 @@ export const DetailsContent: FC<{ id: string }> = ({ id }) => {
   );
 };
 
-export const Details = () => {
-  const { id, authority } = useParams<{
-    id: string;
-    authority: Authority;
-  }>();
-
+export const packageDetailsLoader = async ({ params }: LoaderFunctionArgs) => {
+  const { id, authority } = params;
   if (id === undefined || authority === undefined) {
-    return <Navigate to="/dashboard" />;
+    return redirect("/dashboard");
   }
 
+  try {
+    const packageResult = await getItem(id);
+    if (!packageResult || packageResult?._source?.deleted === true) {
```

Comment:
Wouldn't it be clearer to check:

```ts
if (packageResult && packageResult._source.deleted === true)
```

Reply:
Hmm. I also want to redirect if the packageResult doesn't exist though (if the id is invalid)

Reply:
Gotcha, in that case, resolving